util/builtinobjects/builtinobjects.go

```diff
@@ -405,43 +417,96 @@ func (b *builtinObjects) getNewObjectID(spaceID string, oldID string) (id string
 	return "", err
 }
 
-func (b *builtinObjects) downloadZipToFile(url string) (path string, err error) {
+func (b *builtinObjects) downloadZipToFile(url string) (path string, progress process.Progress, err error) {
```

Comment:
maybe pass progress as parameter? It will reduce a scope of the method

Reply:
Done!